src/Components/Book-list.jsx

```diff
@@ -1,18 +1,59 @@
-
-import BookItem from "./Book-item"
-import { useSelector } from 'react-redux'
+import PropTypes from 'prop-types';
+import BookItem from "./Book-item";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchBooks } from "../redux/book/bookSlice";
+import { useEffect } from "react";
 
 function BookList() {
-    const bookStore = useSelector((state) => state.bookStore)
-  return (
-      <ul>
-          {bookStore.map((book) =>(
-           <BookItem  key={book.item_id} id={book.item_id} {...book} />
-          ))}
-      </ul>
-  )
-}
+  const { bookStore, isLoading, error } = useSelector(
+    (state) => state.bookStore
+  );
+  const dispatch = useDispatch();
 
+  useEffect(() => {
+    dispatch(fetchBooks());
+  }, [dispatch]);
+
+  if (isLoading) {
+    return (
+      <div>
+        <h4>Loading...</h4>
+      </div>
+    );
+  } else if (error) {
+    return (
+      <div>
+        <h4>Error: {error}</h4>
+      </div>
+    );
+  } else if (Object.keys(bookStore).length === 0) {
+    return (
+      <div>
+        <h3>No books found</h3>
+      </div>
+    );
+  } else {
+    const bookData = Object.values(bookStore)[0];
+   
+    return (
+      <div>{
+        bookData.map((book) =>(
```

Comment:
- [ ] Please make sure you adjust your page content in the middle as for now it is overlapping to the right

<img width="1264" alt="1" src="https://github.com/ghermaico135/react-redux-bookstore/assets/77843050/41f9c6e1-c33b-4026-bc9a-6f98f8ad6bf0">